Run ActionSpaceWalker on the new variable free language for WikiTables

pdasigi opened the pull request:
Various changes to existing code, and a script to run the `ActionSpaceWalker` for offline search.

pdasigi commented:
Will rebase after I merge #1832 (waiting for TeamCity to do its thing). @MurtyShikhar you should be able to run the offline search script (`scripts/wikitables/search_for_logical_forms.py`) now. The argparse usage instructions should be self-explanatory.

matt-gardner commented on scripts/wikitables/evaluator.py:
```diff
@@ -0,0 +1,395 @@
+#!/usr/bin/env python
```
This needs to go inside `allennlp`, so that it can be used in models (and so that it gets distributed via pip installs, etc.).  We currently have `squad_eval.py` in `allennlp.common`, and `srl-eval.pl` in `allennlp.tools`.  I'd say this should go in `allennlp.tools` (and `squad_eval.py` should probably be moved there).

pdasigi replied:
Done.

pdasigi commented:
@matt-gardner This should be ready to review. I was waiting to make agenda improvements, but I think that will be an iterative process and can go in a separate PR. Also, this PR is fairly big at this point. The main changes are related to making the `ActionSpaceWalker` run, but there are also other related changes (like adding the official evaluator), and I refactored some bits (made a generic wikitables data util module, and moved squad_eval to tools as you suggested).

The test coverage is low because of the wikitables evaluator. Is there anything we can do there, other than write tests for it?

matt-gardner commented on allennlp/semparse/contexts/table_question_context.py:
```diff
@@ -81,17 +82,27 @@ class TableQuestionContext:
-        self.column_types = {column_index_to_name[column_index]: max(column_type_statistics[column_index],
-                                                                     key=column_type_statistics[column_index].get)
-                             for column_index in column_index_to_name}
+        self.table_data = table_data
```
You're taking an input here, then immediately overriding it below.  Something is off.

pdasigi replied:
Oh, thanks! Fixed it and added some comments.
